tests/rados/test_rados_preempt_scrub.py

```diff
@@ -18,210 +19,236 @@
-                    log.info("Scrubbing is in progress")
-                    break
-                log.info("Waiting for the scrub to start")
-                time.sleep(30)
-                count = count + 1
-                log.info(f"The count is -{count}")
-            total_time = count * 30
-            log.info(
-                f"The total time to start user initiated scrub is  - {total_time} seconds"
+        object_list = rados_object.get_object_list(pool_name)
+        rados_object.run_deep_scrub(pool=pool_name)
+        time.sleep(5)
+        for _ in range(30):
+            tmp_date = subprocess.check_output("date", text=True).strip()
+            for object in object_list:
+                if re.search(r".*object.*", object):
+                    cmd_rewrite = (
+                        f"echo {tmp_date} | rados -p {pool_name} put {object} -"
+                    )
+                    rados_object.client.exec_command(cmd=cmd_rewrite, sudo=True)
+        get_paramter_value(mon_obj, rados_object, pool_name)
+        # Wait time for starting the scheduled scrubbing
+        time.sleep(180)
+
+        # Check for 30 minutes
+        time_end = time.time() + 60 * 30
+        found_flag = False
+        while time.time() < time_end:
+            time.sleep(10)
+            end_time, _ = installer.exec_command(
+                cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'"
             )
-            # check for the scrub is initiated or it is timeout
-            if scrub_status_flag is False:
-                log.error("Scrub is not initiated")
-                return 1
-            log.info("Scrub initiated")
-            time.sleep(30)
-            end_time, _ = installer.exec_command(cmd="sudo date '+%Y-%m-%d %H:%M:%S'")
-            osd_list = []
-            for node in ceph_nodes:
-                if node.role == "osd":
-                    node_osds = rados_object.collect_osd_daemon_ids(node)
-                    osd_list = osd_list + node_osds
-            log.info(f"The number of OSDs in the cluster are-{len(osd_list)}")
-
-            log_osd_count = 0
-            for osd_id in osd_list:
+            end_time = end_time.strip()
+            for osd_id in acting_set:
+                time.sleep(5)
+                log.info(f"Checking the logs at: {osd_id}")
                 if verify_preempt_log(
-                    rados_obj=rados_object,
-                    osd=osd_id,
-                    start_time=init_time,
+                    init_time=init_time,
                     end_time=end_time,
+                    rados_object=rados_object,
+                    osd_id=osd_id,
                     lines=log_lines,
                 ):
                     log.info(f"The preempted lines found at {osd_id}")
-                    log_osd_count = log_osd_count + 1
-                    time.sleep(10)
-                if log_osd_count == 2:
+                    found_flag = True
+                if found_flag:
```

Comment:
Create a backlog task for check for log messages in all the OSDs of the acting set.

Currently we exit even if it's found in 1 osd in the AC.

Reply:
Jira task raised - https://issues.redhat.com/browse/RHCEPHQE-17710